### ISSUE TITLE
lcm_driven_loop_test Failure - Matrix size mismatch: (1 x 6 vs. 1 x 5)

### ISSUE THREAD

**Issue by liangfok** [RobotLocomotion/drake#5936]
# The Error

```
[ RUN      ] LcmDrivenLoopTest.TestLoop
/home/ubuntu/workspace/linux-xenial-clang-experimental/drake/systems/lcm/test/lcm_driven_loop_test.cc:131: Failure
Value of: drake::CompareMatrices(expected.transpose(), logger->data(), 1e-12, drake::MatrixCompareType::absolute)
  Actual: false (Matrix size mismatch: (1 x 6 vs. 1 x 5))
Expected: true
[  FAILED  ] LcmDrivenLoopTest.TestLoop (1104 ms)
```

Example log: https://drake-jenkins.csail.mit.edu/job/linux-xenial-clang-experimental/3902/

# Initial Suspicion

The unit test is using a _real_ LCM instance. Maybe it accidentally received a message that was not intended for it? Can the unit test be modified to use `DrakeMockLcm`?

**Comment by siyuanfeng-tri** [comment]
hmm. probably a lcm flakyness. 
I can probably make it work with fake lcm, but i need to change stuff there.

**Comment by liangfok** [comment]
BTW, it timed out in https://drake-jenkins.csail.mit.edu/job/linux-xenial-clang-experimental/3905/:

```
260/260 Test #172: lcm_driven_loop_test ...............................***Timeout  60.01 sec
```

**Comment by jamiesnape** [comment]
This is blocking moving various Bazel memcheck builds to build cop.

**Comment by jwnimmer-tri** [comment]
At worst, just `tags=['nomemcheck']` with a TODO though, right?

**Comment by siyuanfeng-tri** [comment]
If it's blocking, feel free to take it off CI.

**Comment by jamiesnape** [comment]
> At worst, just tags=['nomemcheck'] with a TODO though, right?

Yes, but for the various issues with combining tags.

**Comment by jamiesnape** [comment]
Would anyone object to marking it `flaky` with a TODO?

**Comment by siyuanfeng-tri** [comment]
Idk what flaky means, but sounds good to me...

**Comment by jamiesnape** [comment]
https://bazel.build/versions/master/docs/be/common-definitions.html#common-attributes-tests:
> Marks test as flaky.
>
> If set, executes the test up to 3 times before being declared as failed. By default this attribute is set to 0 and test is considered to be stable. Note, that use of this attribute is generally discouraged - we do prefer all tests to be stable.

**Comment by m-chaturvedi** [comment]
We have disabled the test for a few sanitizers, we would want to re-enable them after we've sorted this issue. https://github.com/RobotLocomotion/drake/pull/6327

**Comment by hongkai-dai** [comment]
@siyuanfeng-tri could you update the status of this issue?

**Comment by jwnimmer-tri** [comment]
I will dispose of this issue forthwith.

**Comment by jwnimmer-tri** [comment]
Closed as of #11281.
